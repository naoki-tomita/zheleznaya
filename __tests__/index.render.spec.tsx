import { h, render, Component } from "../index";
describe("#render", () => {
  it("should rendering dom.", () => {
    const someString = "someString";
    const Inner: Component = (props: { name: string }, children) => {
      return (
        <div>
          <div>
            child component {props.name}
          </div>
          {children}
          <div>
            child component end
          </div>
        </div>
      );
    }
    const App = () => {
      const list = ["foo", "bar", "hoge"]
      return (
        <div id="root">
          <h1>Hello world</h1>
          <h2 style={{ display: "flex", justifyContent: "center" }}>Style</h2>
          <h3 class="test-foo">className</h3>
          <h4>{someString}</h4>
          <div ref={el => el.innerHTML = "<span>ref innerHTML</span>"} />
          <Inner name="props.name" >Hello <div>world</div></Inner>
          <Inner name="array children">
            <ul>
              {list.map(key => <li>{key}</li>)}
            </ul>
          </Inner>
        </div>
      );
    }
    render(<App />)
    expect(document.body).toMatchSnapshot()
  });

  it("should call callback when button clicked.", () => {
    const spy = jest.fn();
    const App = () => {
      return <button onclick={spy} >click me!</button>
    }
    render(<App />);
    const button = document.body.querySelector("button")!;
    button.click();
    expect(spy).toBeCalled();
  });
});
