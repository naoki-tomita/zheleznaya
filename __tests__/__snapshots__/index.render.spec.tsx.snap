// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render should rendering dom. 1`] = `
<body>
  <div>
    <div
      id="root"
    >
      <h1>
        Hello world
      </h1>
      <h2
        style="display: flex; justify-content: center;"
      >
        Style
      </h2>
      <h3
        class="test-foo"
      >
        className
      </h3>
      <h4>
        someString
      </h4>
      <div>
        <span>
          ref innerHTML
        </span>
      </div>
      <div>
        <div>
          child component 
          props.name
        </div>
        Hello 
        <div>
          world
        </div>
        <div>
          child component end
        </div>
      </div>
      <div>
        <div>
          child component 
          array children
        </div>
        <ul>
          <li>
            foo
          </li>
          <li>
            bar
          </li>
          <li>
            hoge
          </li>
        </ul>
        <div>
          child component end
        </div>
      </div>
    </div>
  </div>
</body>
`;
